### PR TITLE
Implement Unregister APIs for PushNotificationManager

### PR DESCRIPTION
**Pull request by sharath2727**

### CODE REVIEW

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ constexpr std::wstring_view expectedPushServerArgs = L"----WindowsAppRuntimePush
 
 static wil::unique_event g_waitHandleForArgs;
 static winrt::guid g_comServerClsid{ GUID_NULL };
+static bool registering{ false };
```
Comment (pmpurifoy) [inline dev/PushNotifications/PushNotificationManager.cpp:33]: Can this be a member variable instead?

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 2]
```diff
@@ -202,9 +203,17 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                     // Need to recreate the Foreground sinks since channel creation removes the old sink
                     RegisterForegroundSinkHelper();
 
+                    auto channel{ winrt::make<PushNotificationChannel>(channelInfo) };
+                 //   {  };
```
Comment (pmpurifoy) [inline dev/PushNotifications/PushNotificationManager.cpp:207]: Remove this line.

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 3]
```diff
@@ -447,44 +486,95 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
...
+        {
+            try
+            {
+                bool channelExists{ false };
+                {
+                    auto lock{ m_lock.lock_exclusive() };
+                    channelExists = (m_channel != nullptr) ? true : false;
+                }
+
+                if (channelExists)
```
Comment (pmpurifoy) [inline dev/PushNotifications/PushNotificationManager.cpp:522]: This can move into the lock exclusive block above

**Review by pmpurifoy — APPROVED** [review]

File: dev/PushNotifications/PushNotificationManager.cpp (hunk 1 above)
Comment (sharath2727) [reply dev/PushNotifications/PushNotificationManager.cpp:33]: No I think it is fine let this be static.

**Pull request merged**